components/scream/src/physics/p3/atmosphere_microphysics.hpp

```diff
@@ -206,6 +212,29 @@ class P3Microphysics : public AtmosphereProcess
   }; // p3_postamble
   /* --------------------------------------------------------------------------------------------*/
 
+  // Structure for storing local variables initialized using the ATMBufferManager
+  struct Buffer {
```

Comment:
Food for thought: later, we might see if it's beneficial to make a Buffer struct storing values within a column, in order to achieve better data locality. Then store a view of Buffer instead of a single Buffer.

Reply:
I'm guessing this would involve passing the buffer to `shoc_main()` instead of aliasing input/output views in `SHOCMacrophysics::initialize_impl()`?

Side (but maybe related) note: There are a lot of shoc variables labeled `diagnostic output` which are never used outside of `shoc_main()`. What is the point of keeping these around as global variables in internal shoc, instead of having them as local variables int the workspace manager?

Reply:
Not sure about shoc, but in P3 the short answer is yes. I see that diag_outputs contain both Field's views and Buffer's views. That's somewhat of a mix, and prevents us from passing a view_1d<Buffer>. Your 2nd question is probably linked to if/how we can do this. Imho, shoc and p3 (as well as every other atm proc that can be written as a single big kernel) should not have buffer views of length (ncols,...), but rather rely completely on WSM slots. There's no point in allocating (ncols,nlev) views if you don't plan to stick them in the FM (or, as it's the case for Homme, need these views to persist across 2+ kernels).

Reply:
Is this worth focusing on now? I wouldn't mind continuing working on it.

Reply:
I don't think so. Memory access on GPU is relatively fast, so locality has a lesser impact than it may have on CPU. I would add this to v1.1 or v2 stuff.